### PR TITLE
IGL-122: Fixed localization in Pager. Added Provider. Updated depende…

### PR DESCRIPTION
**Pull request by vicky-comeau**
…ncies

### CODE REVIEW

File: packages/Provider/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const base = require('../../jest.config');
+const { name } = require('./package.json');
+
+module.exports = {
+  ...base,
+  name,
+  displayName: name,
```
Comment (fraincs) [inline packages/Provider/jest.config.js:7]: we can remove the trailing comma here

**Review by fraincs — DISMISSED** [review]

File: packages/Datepicker/src/Datepicker.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ import Calendar from "./components/Calendar";
 
 import "./datepicker.scss";
 
-interface Date { utc: string; local: string }
+interface Date { utc: string; local: string | null }
```
Comment (alexasselin008) [inline packages/Datepicker/src/Datepicker.tsx:24]: local: string | null

is that related to the provider or a package update?

File: packages/Pager/src/Pager.tsx [hunk 3]
```diff
@@ -153,7 +159,7 @@ const Pager: React.FunctionComponent<PagerProps> = ({
             </nav>
             <div className="ids-pager__results">
                 {pageSize * currentPage - pageSize + 1}-
-                {Math.min(pageSize * currentPage, totalCount)} of {totalCount}
+                {Math.min(pageSize * currentPage, totalCount)} {stringFormatter.format("of")} {totalCount}
```
Comment (alexasselin008) [inline packages/Pager/src/Pager.tsx:162]: i'm 90% certain that there is a way for the stringFormatter to be used with templates. Looking at react-spectrum, it should be 

in intl file:
```
  "pageRange": "{count} of {totalCount}"
```

in js
```
stringFormatter.format('pageRange', {count: 1, totalCount: 4});
```

Comment (alexasselin008) [reply packages/Pager/src/Pager.tsx:162]: same comment for                      

                          : `${stringFormatter.format("goToPage")} ${pageNumber}`

File: packages/Datepicker/src/Datepicker.tsx (hunk 2 above)
Comment (vicky-comeau) [reply packages/Datepicker/src/Datepicker.tsx:24]: It's most likely related to a package update. It complained without it.

File: packages/Pager/src/Pager.tsx (hunk 3 above)
Comment (alexasselin008) [inline packages/Pager/src/Pager.tsx:162]: Might not be that bad in this scenario, but i think we try to offer translation for aria-labels AND offer defaults that make sense for string that are visually rendered. But the consumers should always be able to override the strings. 

for instance, in this case, a pager should offer a method that takes 3 parameters and return a string or ReactNode, and we offer the default value which would be the string template

Comment (vicky-comeau) [reply packages/Pager/src/Pager.tsx:162]: yes it can be, but I figured they were positioned in the same place so I didn't add it. If you think I should, then I'll fix it.

Comment (alexasselin008) [reply packages/Pager/src/Pager.tsx:162]: React-spectrum for instance offer a "valueLabel" props

https://react-spectrum.adobe.com/react-aria/Meter.html#props

valueLabel - ReactNode- The content to display as the value's label (e.g. 1 of 4).
showValueLabel - boolean- Whether the value's label is displayed. True by default if there's a label, false by default if not.

Comment (vicky-comeau) [reply packages/Pager/src/Pager.tsx:162]: so from what I understand, you want me to add a prop like renderResultsText that passes the pageSize, currentPage and totalCount?
If not set, it defaults to our default message?

File: packages/Provider/src/Provider.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import * as React from "react";
+
+import { I18nProvider, type I18nProviderProps, useLocale, useLocalizedStringFormatter } from "react-aria";
+
+import "./provider.scss";
```
Comment (alexasselin008) [inline packages/Provider/src/Provider.tsx:5]: remove this line, there is no scss

File: packages/Provider/src/intl/en-US.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "testMessage": "test message"
```
Comment (alexasselin008) [inline packages/Provider/src/intl/en-US.json:2]: Hmm i don't like the idea of having localized test messages. those folders are being shipped to npm, and if we were to add non-test strings, then both production and tests strings would be loaded by client applications

File: packages/Provider/src/Provider.tsx (hunk 4 above)
Comment (vicky-comeau) [reply packages/Provider/src/Provider.tsx:5]: I originally had done this, but it kept complaining that there was no provider.css in the dist file... I removed all references to it as far as I know, it might have been because we use it in the decorator..

File: packages/Provider/src/intl/en-US.json (hunk 5 above)
Comment (alexasselin008) [inline packages/Provider/src/intl/en-US.json:2]: I wouldn't add test messages in the intl folder. We might end up with a mix and match of real and tests values, and they are all going to be sent to the users. I suggest you create a local object in your test file instead

File: packages/Provider/src/Provider.tsx (hunk 4 above)
Comment (alexasselin008) [reply packages/Provider/src/Provider.tsx:5]: and delete the .scss file as well. No need to keep stuff just in case

File: packages/Pager/package.json [hunk 6]
```diff
@@ -24,8 +24,9 @@
     "build": "rollup -c rollup.config.js"
   },
   "dependencies": {
+    "@igloo-ui/provider": "^0.0.0",
```
Comment (alexasselin008) [inline packages/Pager/package.json:27]: it has to be a peer dependency and dev dependency

File: packages/Pager/src/Pager.tsx (hunk 3 above)
Comment (alexasselin008) [reply packages/Pager/src/Pager.tsx:162]: We should do it the right way ya. to start getting used to it

Comment (alexasselin008) [reply packages/Pager/src/Pager.tsx:162]: They have all the informations already, so the function is overkill.

add to the interface
```
valueLabel?: ReactNode
```

and we set it's default value in our codebase if it's null
```
valueLabel = stringFormatter.format("pageRange", { minRange: pageSize * currentPage - pageSize + 1, maxRange: Math.min(pageSize * currentPage, totalCount), totalCount: totalCount })
```

you can chose variable name that makes more sense in the context (minRange,maxRange,totalCount)

File: packages/Provider/src/Provider.tsx (hunk 4 above)
Comment (alexasselin008) [reply packages/Provider/src/Provider.tsx:5]: I'll investigate, there is no reason this file should exist.

**Review by alexasselin008 — APPROVED** [review]
